### ISSUE TITLE
Squiz.Functions.MultiLineFunctionDeclaration reports error if php attributes are declared

### ISSUE THREAD

**Issue by vincent-paqt** [PHPCSStandards/PHP_CodeSniffer#608]
## Describe the bug

When running phpcs with a class where multiple php attributes are declared  `Squiz.Functions.MultiLineFunctionDeclaration.OneParamPerLine` reports an error. 

### Code sample
```php
class MultiLineFunctionDeclarationFixesAttributes
{
    public function __construct(
        #[WithCast(DateTimeInterface::class), WithTransformer(DateTimeInterface::class, format: 'Y-m-d')]
        public readonly string $transferDate,
    ){
    }
}
```

### Custom ruleset
```xml
<?xml version="1.0"?>
<ruleset name="My Custom Standard">
  <rule ref="Squiz.Functions.MultiLineFunctionDeclaration"/>
</ruleset>
```

### To reproduce
1. Create a file called `test.php` with the code sample above...
2. Run `phpcs test.php ...`
3. See error message displayed
```text
----------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------------------------------------------------------------------------------------
 5 | ERROR | [x] Multi-line function declarations must define one parameter per line (Squiz.Functions.MultiLineFunctionDeclaration.OneParamPerLine)
----------------------------------------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------------------------------------------------------------------------------------
```

## Expected behavior

Skip Squiz.Functions.MultiLineFunctionDeclaration when its between `#[` and `]`

## Versions (please complete the following information)

|                                       |                                                                                                           |
|--------------------------|----------------------------------------------------------------------  |
| Operating System           | Alpine Linux v3.19                                                       |
| PHP version                    | 8.3.7                                                                                     |
| PHP_CodeSniffer version | squizlabs/php_codesniffer 3.10.2                                                                           |
| Standard                         | PSR12, custom                                                      |
| Install type                      | composer |

## Please confirm

- [x] I have searched the issue list and am not opening a duplicate issue.
- [x] I have read the [Contribution Guidelines](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/master/.github/CONTRIBUTING.md) and this is not a [support question](https://github.com/PHPCSStandards/PHP_CodeSniffer/discussions).
- [x] I confirm that this bug is a bug in PHP_CodeSniffer and not in one of the external standards.
- [x] I have verified the issue still exists in the `master` branch of PHP_CodeSniffer.

**Comment by jrfnl** [comment]
Thanks @vincent-paqt for reporting this. I've been able to reproduce this. Of the three code sampes you provided, only the first one is relevant, the rest was unrelated code, so I've taken the liberty to remove this so as not to confuse people.

Based on some additional testing, this bug only happens with "multi-attributes", i.e. multiple comma separated attribute classes within a single set of attribute braces `#[...]`.
* "Single attributes" do not trigger the error.
* Multiple "single attributes" do not trigger the error.

When the fixer is run, it can also clearly be seen where the sniff gets confused:
```php
class MultiLineFunctionDeclarationFixesAttributes
{
    public function __construct(
        #[WithCast(DateTimeInterface::class),
 WithTransformer(DateTimeInterface::class, format: 'Y-m-d')]
        public readonly string $transferDate,
    ) {
    }
}
```

Other observations: while the Squiz sniff extends a PEAR sniff, this bug only occurs in the Squiz sniff. The PEAR sniff is not affected.

**Comment by jrfnl** [comment]
@vincent-paqt PR #609 should fix this. Testing appreciated.

**Comment by vincent-paqt** [comment]
@jrfnl Thanks for the quick response and fix! I've tested #609 and it fixes the bug. I don't get the errors anymore.
